### PR TITLE
Move assignments with init or record to a separate method

### PR DESCRIPTION
**Pull request by riyafa**
## Purpose

Related issue: https://github.com/ballerina-platform/ballerina-lang/issues/23376

## Approach
To prevent the method too large issue, this fix moves the assignment statements with init or record expressions to a separate method, thus making the init method smaller

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by riyafa** [comment]
I thought it would be done manually. But I can try to create a module that will make the init function bigger which fails during build in the previous versions and passes in this?

**Comment by Kishanthan** [comment]
> I thought it would be done manually. But I can try to create a module that will make the init function bigger which fails during build in the previous versions and passes in this?

Yes, lets add a test case that verifies this fix.

**Comment by riyafa** [comment]
The difficult part of adding the test is getting the init method to be too large within the 50 statements limit. One thing we can do to write a test is to make this 50 statements limit configurable. @Kishanthan WDYT?

**Comment by Kishanthan** [comment]
> The difficult part of adding the test is getting the init method to be too large within the 50 statements limit. One thing we can do to write a test is to make this 50 statements limit configurable. @Kishanthan WDYT?

Yes, that is an option.
> 
> We cannot use the project from #23376 directly because it requires so many dependencies and hence will make this repo larger if we use it.
> 
> Between this PR is for the 1.1.x branch while the plan is to work on a proper implementation for the master branch

But we can add this fix to master as well and work on the optimization improvement. Shall we push this fix to both 1.2.x and master branches?

File: tests/jballerina-integration-test/src/test/resources/testng.xml [hunk 1]
```diff
@@ -294,12 +294,12 @@
...
-            <class name="org.ballerinalang.test.packaging.ModulePushTestCase"/>
+<!--            <class name="org.ballerinalang.test.packaging.ModulePushTestCase"/>-->
             <!--            <class name="org.ballerinalang.test.packaging.PackagingNegativeTestCase"/>-->
-            <class name="org.ballerinalang.test.packaging.PackagingTestCase"/>
-            <class name="org.ballerinalang.test.packaging.LockFileTestCase"/>
+<!--            <class name="org.ballerinalang.test.packaging.PackagingTestCase"/>-->
+<!--            <class name="org.ballerinalang.test.packaging.LockFileTestCase"/>-->
             <!--            <class name="org.ballerinalang.test.packaging.NativePackagingTestCase"/>-->
-            <class name="org.ballerinalang.test.packaging.PathDependencyTestCase"/>
+<!--            <class name="org.ballerinalang.test.packaging.PathDependencyTestCase"/>-->
```
Comment (Kishanthan) [inline tests/jballerina-integration-test/src/test/resources/testng.xml:302]: Are these packaging tests already failing in this 1.1.x branch? @hevayo @keizer619

**Comment by riyafa** [comment]
To see if changing the size affects the performance. Did a small perf test with the  project in issue #23376 and got the following results:
With method size 50:     53.75s    
With method size 2:        49.85s

Tested with another file that has large records initialized and results are as follows:
With method size 50:     10.11s
With method size 2:        10.01s

Runtime for the same:
With method size 50:     347ms
With method size 2:        354ms

File: tests/jballerina-integration-test/src/test/resources/testng.xml (hunk 1 above)
Comment (riyafa) [reply tests/jballerina-integration-test/src/test/resources/testng.xml:302]: These tests fail when pulling or pushing to central.

Comment (Kishanthan) [reply tests/jballerina-integration-test/src/test/resources/testng.xml:302]: @keizer619 @hevayo , is this expected with 1.1.x branch? Shall we go ahead with disabling the tests?

Comment (Kishanthan) [reply tests/jballerina-integration-test/src/test/resources/testng.xml:302]: This is ok as per @hevayo 
"tests are failing probably due to lets encrypt migration of the staging server. And prior 1.2 clients are not compatible with that."

**Pull request merged**